crates/runtime/README.md

```diff
@@ -0,0 +1,662 @@
+memory management system works closely with the `VMLogic` struct to provide safe
+memory access to the guest Wasm code. The `VMLogic` struct includes methods for
+reading from and writing to the Wasm module's memory, ensuring that all memory
+accesses are bounds-checked and secure.
+
+The memory management system also plays a crucial role in the implementation of
+host functions. Many host functions, such as those for reading and writing data,
+interacting with registers, or performing I/O operations, require careful
+management of memory. The system ensures that these operations are performed
+safely, preventing buffer overflows, use-after-free vulnerabilities, and other
+memory-related security issues.
+
+Error handling is another important aspect of the memory management system. When
+memory-related errors occur, such as attempts to allocate more memory than
+allowed or access memory outside of bounds, the system generates appropriate
+error responses. These errors are then propagated through the runtime, allowing
+for proper handling and reporting of memory-related issues.
+
+The memory management system also considers performance implications. By
+carefully tuning memory allocations and access patterns, it aims to provide
+efficient memory usage while maintaining strict security controls. This balance
+is crucial for ensuring that the Calimero runtime can execute Wasm modules with
+good performance while still providing a secure and controlled environment.
+
+### Storage
+
+The `store.rs` file defines the `Storage` trait and provides an in-memory
+implementation:
+
+  - Key-value storage interface
+  - Operations: `get`, `set`, `has`
+
+The storage system in the Calimero runtime maintains state and facilitates data
+persistence across Wasm module executions. Its primary purpose is to provide a
+controlled, secure, and flexible mechanism for Wasm modules to store and
+retrieve data, enabling stateful applications and complex smart contract-like
+functionality within the runtime environment.
+
+At the core of the storage system is the `Storage` trait, defined in the
+`store.rs` file. This trait establishes a standard interface for storage
+operations, including methods for getting, setting, and checking the existence
+of key-value pairs. By defining storage as a trait, the Calimero runtime
+achieves a high degree of flexibility, allowing for different storage
+implementations to be used interchangeably depending on the specific needs of
+the application or deployment environment.
+
+The most basic implementation provided in the runtime is the `InMemoryStorage`
+struct, which uses a `BTreeMap` to store key-value pairs in memory. This
+implementation is particularly useful for testing, development, and scenarios
+where persistent storage is not required. However, the trait-based approach
+allows for easy integration of more sophisticated storage solutions, such as
+disk-based storage, distributed databases, or blockchain-specific storage
```

Comment:
what is meant by "blockchain-specific storage"?

Reply:
Xabi has been talking to me about the possibilities of linking into blockchain systems from our storage, so I thought I would mention it here. Too much...? Too tenuous?

Reply:
Hm, I.. think we can remove it here